src/core/scope.c

```diff
@@ -353,35 +359,87 @@ static void scope_enter_signal(Scope *s, ScopeState state, ScopeResult f) {
         scope_enter_dead(s, SCOPE_FAILURE_RESOURCES);
 }
 
-static int scope_start(Unit *u) {
-        Scope *s = SCOPE(u);
+static int scope_enter_chown_cgroup(Scope *s) {
+        Unit *u = UNIT(s);
+        pid_t pid;
         int r;
 
         assert(s);
 
-        if (unit_has_name(u, SPECIAL_INIT_SCOPE))
-                return -EPERM;
+        (void) unit_realize_cgroup(u);
+        (void) unit_reset_accounting(u);
 
-        if (s->state == SCOPE_FAILED)
-                return -EPERM;
+        r = scope_arm_timer(s, usec_add(now(CLOCK_MONOTONIC), 30*USEC_PER_SEC));
+        if (r < 0)
+                return r;
```

Comment:
This hard-coded timeout looks fishy.
Couldn't this be charged against `JobRunningTimeoutSec=`? (I know it's not the same, especially diverts from service start, however, it'd reuse the existing mechanisms.)